File: Programming 2/Labs/Store in Python/Store.py
from Product import *
from CashRegister import *

class Store:
    def __init__(self):
        self.cash_register = CashRegister()
        self.sticky_tape = Product("Sticky tape", 200, 2.99)

    def help_menu(self):
        print("Menu options\ns = sell\nr = restock\nv = view stock\nc = view cash\nx = exit")
    
    def sell(self):
        amount = int(input("Number: "))
        if amount > self.sticky_tape.get_stock():
            print("Not enough stock")
        else:
            self.cash_register.add(self.sticky_tape.sell(amount))
    def restock(self):
        amount = int(input("Number: "))
        self.sticky_tape.restock(amount)
    
    def view_stock(self):
        print(self.sticky_tape.to_string())
    
    def view_cash(self):
        print(self.cash_register.to_string())
    
    def use(self):
        decision = 0
        while decision != 'x':
            decision = str(input("Choice (s/r/v/c/x): "))
            if decision =='s':
                self.sell()
            if decision == 'r':
                self.restock()
            if decision == 'c':
                self.view_cash()
            if decision == 'v':
                self.view_stock()
            if decision == '?':
                self.help_menu()
        print("Done")
store = Store()
store.use()

File: Programming 2/Labs/Store/Store.java
/**
 * The store has and sells one product: Sticky tape.
 *
 * You can sell and restock a product, view stock and view cash.
 */
public class Store {
    private CashRegister cash_register;
    private Product sticky_tape;
    
    public static void main(String[] args){
        Store store = new Store();
        store.use(); 
    }
    // insert main method here
    // insert 2 fields here
    // insert 1 constructor here
    public Store() {
        this.cash_register = new CashRegister(); 
        this.sticky_tape = new Product("Sticky tape", 200, 2.99); 
    }
    public void use() {
        char choice;
        while ((choice = readChoice()) != 'x') {
            switch (choice) {
            case '?':
                help();
                break;
            case 'v':
                viewStock();
                break;
            case 's':
                sell();
                break;
            case 'r':
                restock();
                break;
            case 'c':
                viewCash();
                break;
            default:
                break;
            }
        }
        System.out.println("Done");
    }

    private void sell() {
        System.out.print("Number: ");
        int amount = In.nextInt();
        if (amount > sticky_tape.get_stock()) {
            System.out.println("Not enough stock");
        }
        else {
            this.cash_register.add(this.sticky_tape.sell(amount));
        }
    }

    private void restock() {
        System.out.print("Number: ");
        int amount = In.nextInt();
        this.sticky_tape.restock(amount);
    }

    private void viewStock() {
        System.out.println(this.sticky_tape.toString());
    }

    private void viewCash() {
        System.out.println(this.cash_register.toString());
    }

    private void help() {
        System.out.println("Menu options");
        System.out.println("s = sell");
        System.out.println("r = restock");
        System.out.println("v = view stock");
        System.out.println("c = view cash");
        System.out.println("x = exit");
    }
    public char readChoice() {
        System.out.print("Choice (s/r/v/c/x): ");
        return In.nextChar();
    }
}
